domain/secret/service/exportimport.go

```diff
@@ -160,19 +166,25 @@ func (s *SecretService) ImportSecrets(ctx context.Context, modelSecrets *SecretE
 			return errors.Annotatef(err, "saving secret %q", md.URI.ID)
 		}
 		for _, sc := range modelSecrets.Consumers[md.URI.ID] {
-			if err := s.secretState.SaveSecretConsumer(ctx, md.URI, sc.Accessor.ID, &coresecrets.SecretConsumerMetadata{
-				Label:           sc.Label,
-				CurrentRevision: sc.CurrentRevision,
-			}); err != nil {
+			err = s.secretState.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
```

Comment:
Similarly, the entire import operation should be in a single txn. It doesn't matter so much for import since the model is not usable until the import is done, and so there won't be concurrent access. But we should still do it.